### PR TITLE
ROC-5726-Fix-Duplicate-Event-For-Upload-Documents

### PR DESCRIPTION
**Pull request by karthikkannankainos**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/ROC-5726

### Change description ###

Fix-Duplicate-Event-For-Upload-Documents

### CODE REVIEW

File: src/test/java/uk/gov/hmcts/cmc/claimstore/events/operations/UploadOperationServiceTest.java [hunk 1]
```diff
@@ -27,42 +25,36 @@
...
     @Test
     public void shouldUploadDocument() {
-        //Given
-        UploadOperationService uploadOperationService = new UploadOperationService(documentsService,
-            eventsStatusService);
         //when
         when(documentsService.uploadToDocumentManagement(any(), eq(AUTHORISATION), eq(CLAIM))).thenReturn(CLAIM);
 
-        uploadOperationService.uploadDocument(CLAIM, AUTHORISATION, pinLetterClaim, CLAIM_ISSUE_RECEIPT_UPLOAD);
+        uploadOperationService.uploadDocument(CLAIM, AUTHORISATION, pinLetterClaim);
```
Comment (dharmendrak) [inline src/test/java/uk/gov/hmcts/cmc/claimstore/events/operations/UploadOperationServiceTest.java]: pin letter claim is confusing, please change to sealed claim or  claim receipt

File: src/test/java/uk/gov/hmcts/cmc/claimstore/events/operations/UploadOperationServiceTest.java [hunk 2]
```diff
@@ -27,42 +25,36 @@
...
-        UploadOperationService uploadOperationService = new UploadOperationService(documentsService,
-            eventsStatusService);
+    public void shouldNotUpdateFlagWhenUploadDocumentFails() {
         //when
         when(documentsService.uploadToDocumentManagement(eq(pinLetterClaim), eq(AUTHORISATION), eq(CLAIM)))
             .thenThrow(new DocumentManagementException("bad files"));
 
         try {
-            uploadOperationService.uploadDocument(CLAIM, AUTHORISATION, pinLetterClaim, CLAIM_ISSUE_RECEIPT_UPLOAD);
+            uploadOperationService.uploadDocument(CLAIM, AUTHORISATION, pinLetterClaim);
```
Comment (dharmendrak) [inline src/test/java/uk/gov/hmcts/cmc/claimstore/events/operations/UploadOperationServiceTest.java]: here as well as above

Comment (dharmendrak) [reply src/test/java/uk/gov/hmcts/cmc/claimstore/events/operations/UploadOperationServiceTest.java]: actioned

File: src/test/java/uk/gov/hmcts/cmc/claimstore/events/operations/UploadOperationServiceTest.java (hunk 1 above)
Comment (dharmendrak) [reply src/test/java/uk/gov/hmcts/cmc/claimstore/events/operations/UploadOperationServiceTest.java]: actioned

File: src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/CoreCaseDataService.java [hunk 3]
```diff
@@ -78,6 +82,19 @@
     private final JobSchedulerService jobSchedulerService;
     private final CCDCreateCaseService ccdCreateCaseService;
 
+    private static BiFunction<ClaimSubmissionOperationIndicators, ClaimDocumentType, ClaimSubmissionOperationIndicators>
+        updateClaimSubmissionIndicatorByDocumentType = (indicator, docType) -> {
```
Comment (sabahirfan) [inline src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/CoreCaseDataService.java]: Can we please add unit test for this?

Comment (sabahirfan) [reply src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/CoreCaseDataService.java]: Suggestion: BiFunction usage here doesn't look a right choice here. A good candidate for its usage would be the assignment target for a lambda expression or method reference.

**Pull request merged**